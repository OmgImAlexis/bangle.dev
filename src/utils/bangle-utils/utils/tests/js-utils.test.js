import { matchAllPlus } from '../js-utils';

describe('matchAllPlus', () => {
  test('works when match', () => {
    const result = matchAllPlus(/foo[a-z]*/g, 'baseball  foozball');
    expect(result.map((r) => r.matchedStr)).toMatchInlineSnapshot(`
      Array [
        "baseball  ",
        "foozball",
      ]
    `);
    expect(result).toMatchSnapshot();
  });

  test('works when direct match', () => {
    const result = matchAllPlus(/foo[a-z]*/g, 'foozball');
    expect(result.map((r) => r.matchedStr)).toMatchInlineSnapshot(`
      Array [
        "foozball",
      ]
    `);
    expect(result).toMatchSnapshot();
  });

  test('works when no match', () => {
    const result = matchAllPlus(/foo[a-z]*/g, 'baseball  boozball');
    expect(result.map((r) => r.matchedStr)).toMatchInlineSnapshot(`
      Array [
        "baseball  boozball",
      ]
    `);
    expect(result.every((r) => r.match === false)).toBe(true);
    expect(result).toMatchSnapshot();
  });

  test('works with multiple matches', () => {
    let result = matchAllPlus(
      /foo[a-z]*/g,
      'baseball  football foosball gobhi',
    );
    expect(result.map((r) => r.matchedStr)).toMatchInlineSnapshot(`
      Array [
        "baseball  ",
        "football",
        " ",
        "foosball",
        "gobhi",
      ]
    `);
  });

  test('works with multiple matches', () => {
    const result = matchAllPlus(
      /foo[a-z]*/g,
      'baseball  football gobhi tamatar foosball',
    );
    expect(result.map((r) => r.matchedStr)).toMatchInlineSnapshot(`
      Array [
        "baseball  ",
        "football",
        " gobhi tamatar ",
        "foosball",
      ]
    `);
    expect(result.map((r) => r.match)).toMatchInlineSnapshot(`
      Array [
        false,
        true,
        false,
        true,
      ]
    `);
  });
});
