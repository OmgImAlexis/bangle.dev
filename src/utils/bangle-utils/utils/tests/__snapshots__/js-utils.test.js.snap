// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchAllPlus works when direct match 2`] = `
Array [
  Object {
    "end": 8,
    "match": true,
    "matchedStr": "foozball",
    "start": 0,
  },
]
`;

exports[`matchAllPlus works when match 2`] = `
Array [
  Object {
    "end": 10,
    "match": false,
    "matchedStr": "baseball  ",
    "start": 0,
  },
  Object {
    "end": 18,
    "match": true,
    "matchedStr": "foozball",
    "start": 10,
  },
]
`;

exports[`matchAllPlus works when no match 2`] = `
Array [
  Object {
    "end": 18,
    "match": false,
    "matchedStr": "baseball  boozball",
    "start": 0,
  },
]
`;
